expertise/execute_expertise.py

```diff
@@ -0,0 +1,177 @@
+from pathlib import Path
+import openreview
+from .create_dataset import OpenReviewExpertise
+from .dataset import ArchivesDataset, SubmissionsDataset, BidsDataset
+from .config import ModelConfig
+
+# Move run.py functionality to a function that accepts a config dict
+def execute_expertise(config_file=None, model_config=None):
+
+    if model_config:
+        config = model_config
+    elif config_file:
+        config = ModelConfig(config_dict=config_file)
+    else:
+        raise Exception('Must provide either a model config or a dictionary of config parameters')
+    
+    archives_dataset = ArchivesDataset(archives_path=Path(config['dataset']['directory']).joinpath('archives'))
+    if Path(config['dataset']['directory']).joinpath('submissions').exists():
```

Comment:
I don't think the request.body should define this type of parameter. We have to internally decide which is the directory to store the csv files. 

I think the request body should define a subset of parameters supported in config.json

Reply:
Yeah, this is something that I've been thinking about. What's currently implemented now is - the config gets pre-processed before being executed so that all directories are nested under it's own private directory. So for example if `config['dataset']['directory'] = '/test'`, this would be changed to `config['dataset']['directory'] = 'job_id/test'`:

https://github.com/openreview/openreview-expertise/blob/a86bddea5e48d9efd20d00f8fe1a0f1f1be7e0f8/expertise/service/or_queue.py#L89-L95

But since it doesn't matter to the user what kind of structure the server's filesystem has, I was thinking of just overriding this completely.

I want to avoid modifying the code in the `execute_expertise` function, as the way I have changed the repo, this code is shared with the command line execution:
https://github.com/openreview/openreview-expertise/blob/a86bddea5e48d9efd20d00f8fe1a0f1f1be7e0f8/expertise/run.py#L13-L14

Reply:
I think what you can do is to define a default ModelConfig object where you set all the directories and parameters and in the Flask controller you override some of them from the request body. I can say the parameters that can be sent in the body are:

- name (this must be a mandatory field)
- match_group (this must be a mandatory field)
- paper_invitation (this must be a mandatory field)
- exclusion_inv (optiona, default: None)
- model (optiona, default: SPECTER+MFR)
- model_params.use_title (optional, default: true)
- model_params.use_abstract  (optional, default: true)
- model_params.sparse_value (optional, default: None(?))
- model_params.average_score (optional, default: false)
- model_params.max_score (optional, default: true)
- model_params.skip_specter (optional: default: false)

@carlosmondra please take a look

So I think we don't need to follow the same format of config.json to define the request body parameters because the config model will be based on some internal options and user options so they must be translated anyway.

Reply:
There are some params that are model specific, so I'm not sure if we should throw an exception or just ignore them when parameters are passed that do not correspond to the model that is being used.
We cannot have max_score and average_score at the same time.

name (this must be a mandatory field) <--- I'm not sure if this is even necessary anymore. It depends what we want to use it for. A default name can be used if the param.
model (optional, default: SPECTER+MFR)
model_params.use_title (optional, default: true) <--- I'm not sure where this option is used other than BM25
model_params.use_abstract (optional, default: true) <--- I'm not sure where this option is used other than BM25
model_params.sparse_value (optional, default: None(?))
model_params.average_score (optional, default: false)
model_params.max_score (optional, default: true)
model_params.skip_specter (optional: default: false) <--- Not sure what this means. Maybe it's the separation we talked about between computing scores and computing embeddings